src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -223,22 +229,20 @@ missingAlternative env mn ca uncovered
   mcases = missingCases env mn uncovered ca
 
 -- |
--- Main exhaustivity checking function
+-- Main exhaustivity checking function.
 -- Starting with the set `uncovered = { _ }` (nothing covered, one `_` for each function argument),
 -- it partitions that set with the new uncovered cases, until it consumes the whole set of clauses.
 -- Then, returns the uncovered set of case alternatives.
 -- 
 checkExhaustive :: forall m. (MonadWriter MultipleErrors m) => Environment -> ModuleName -> [CaseAlternative] -> m ()
-checkExhaustive env mn cas = makeResult . first nub $ foldl' step ([initial], (pure True, [])) cas
+checkExhaustive env mn cas = makeResult . first (nub . take maxCases) $ foldr step ([initial], (pure True, [])) cas
```

Comment:
Ups, sorry about that. Yes, you're right. I've changed it to `foldl'`.
I also have an example that blows this implementation up, I just write it here:

```purescript
module Main where
data ABC = A | B | C | D | E | F | G | H | I | J | K | L | M | N | O | P | Q | R | S | T | U | V | W | X | Y | Z | AA | BB | CC | DD | EE | FF | GG | HH | II | JJ | KK | LL | MM | NN | OO | PP | QQ | RR | SS | TT | UU | VV | WW | XX | YY | ZZ

f AA _ B _ C A = A
f B _ BB _ C X = A
f _ AA X X Z ZZ = A
```

I'm working on this right now. I'll let you know how it goes.